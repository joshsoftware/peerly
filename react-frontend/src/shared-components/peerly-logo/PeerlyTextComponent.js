import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const PeerlyText = styled.div`
  color: ${({ theme }) =>
    theme === "dark" ? "var(--white)" : "var(--atomic)"};
  background-color: ${({ theme }) =>
    theme === "dark" ? "var(--atomic)" : "var(--white)"};
  font-size: ${({ fontSize }) => fontSize};
`;

const PeerlyTextComponent = ({ theme, fontSize }) => (
  <PeerlyText
    theme={theme}
    fontSize={fontSize}
    data-testid="PeerlyTextComponents"
  >
    Peerly
  </PeerlyText>
);

PeerlyTextComponent.defaultProps = {
  theme: "dark",
  fontSize: "30px",
};

PeerlyTextComponent.propTypes = {
  theme: PropTypes.oneOf(["dark", "light"]),
  fontSize: PropTypes.string,
};

export default React.memo(PeerlyTextComponent);
