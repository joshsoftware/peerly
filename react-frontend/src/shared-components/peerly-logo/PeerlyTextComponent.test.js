import React from "react";
import { render } from "@testing-library/react";

import "setupTests";
import PeerlyTextComponent from "shared-components/peerly-logo/PeerlyTextComponent";

it("should render PeerlyTextComponent", () => {
  const { asFragment } = render(
    <PeerlyTextComponent theme="dark" fontSize="48px" />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("should test color,fontsize and background of Peerly text", () => {
  const { getByTestId } = render(
    <PeerlyTextComponent theme="dark" fontSize="48px" />
  );
  expect(getByTestId("PeerlyTextComponents")).toHaveStyleRule(
    "color",
    "var(--white)"
  );
  expect(getByTestId("PeerlyTextComponents")).toHaveStyleRule(
    "font-size",
    "48px"
  );
  expect(getByTestId("PeerlyTextComponents")).toHaveStyleRule(
    "background-color",
    "var(--atomic)"
  );
});
