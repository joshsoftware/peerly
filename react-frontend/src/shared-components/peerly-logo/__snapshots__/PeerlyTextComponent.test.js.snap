// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PeerlyTextComponent 1`] = `
<DocumentFragment>
  .c0 {
  color: var(--white);
  background-color: var(--black);
  font-size: 48px;
}

<div
    class="c0"
    data-testid="PeerlyTextComponents"
    font-size="48px"
  >
    Peerly
  </div>
</DocumentFragment>
`;
